### ISSUE TITLE
Error remounting the path: invalid argument, singularity 2.5.1-dist on centOS.

### ISSUE THREAD

**Issue by gabrielmartinh** [apptainer/singularity#1524]
## Version of Singularity:

Singularity 2.5.1-dist
CentOS Linux release 7.2.1511 (Core) 
Kernel: 3.10.0-327.3.1.el7.x86_64

### Expected behavior
After executing "singularity shell container.img" without any other flag the expected behavior is that it provides a shell inside the container.
When executing "singularity exec --nv -c container.img" the expected behavior is that it mounts the nvidia drivers inside the container image such that it can run gpu applications.

### Actual behavior
When executing "singularity shell container.img" without any other special flag it throws an error: 

ERROR  : There was an error remounting the path /etc/localtime: Invalid argument
and it aborts.

When executing "singularity exec --nv -c container.img" it also throws a similar an error:

ERROR  : There was an error remounting the path /bin/nvidia-smi: Invalid argument

If the --nv flag is removed it also shows a similar error:

ERROR  : There was an error remounting /opt/singularity/2.5.1/var/singularity/mnt/session/passwd to /opt/singularity/2.5.1/var/singularity/mnt/final/etc/passwd: Invalid argument

In general singularity 2.5.1-dist is failing in this particular system, the error is the same but depending on the flags and the type of operation it fails to  remount one file or another...

Note: singularity 2.4.2 was working fine in this particular system. 

### Steps to reproduce behavior

Try to run singularity 2.5.1 in a similar enviroment with the same OS and kernel version.

**Comment by intentions** [comment]
I was encountering this problem as well.  the /etc/resolv.conf file wasn't properly binding into the container, showing a size of zero (and causing name lookups to fail).  I applied the patch and saw that the /etc/resolv.conf file had a size, and was able to perform name lookups

**Comment by gabrielmartinh** [comment]
Hi, is it the patch already included in the 2.5.1 release? I updated singularity but it still does not work for me...

**Comment by GodloveD** [comment]
Hi @gabrielmartinh, no this patch is not yet included in a release.  It's been merged into the release-2.5 branch.  To test it, please do something like the following.  

```
git clone https://github.com/singularityware/singularity.git
cd singularity 
git fetch
git checkout release-2.5
./autogen.sh
./configure --prefix /home/$USER/sing-patch
make && sudo make install
~/sing-patch/bin/singularity shell --nv docker://ubuntu
```

**Comment by dtrudg** [comment]
The 2.5 series was merged. Also some resolv.conf and other mount issues mentioned have been addressed in recent 3.x releases.  Please open a new issue if there is a problem on current Singularity 3.5. Thanks!
